src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -3838,11 +3838,29 @@ private static bool CanReuseParameter(CSharp.Syntax.ParameterSyntax parameter)
                 type = null;
                 name = this.EatToken(SyntaxKind.ArgListKeyword);
             }
-            var exclamation = this.CurrentToken.Kind == SyntaxKind.ExclamationToken ? this.EatToken(SyntaxKind.ExclamationToken) : null;
+            SyntaxToken exclamation = null;
+            SyntaxToken equals = null;
+            if (this.CurrentToken.Kind == SyntaxKind.ExclamationToken)
+            {
+                exclamation = this.EatToken(SyntaxKind.ExclamationToken);
+            }
+            else if (this.CurrentToken.Kind == SyntaxKind.ExclamationEqualsToken)
+            {
+                var notEq = this.EatToken();
+                exclamation = new SyntaxToken(SyntaxKind.ExclamationToken);
+                equals = new SyntaxToken(SyntaxKind.EqualsToken);
+                notEq = AddLeadingSkippedSyntax(notEq, SyntaxFactory.MissingToken(SyntaxKind.WhitespaceTrivia));
```

Comment:
1. trivia is not a token.  so `.MissingToken(WhitespaceTrivia)` is not a good idea.  
2. if you are synthesizing tokens here, you must ensure they have the right trivia on them.

Reply:
Is there a missingTrivia function? And how do I transfer trivia between tokens?

Reply:
There is no missing trivia here :).  The lexer spat out `leading-trivia!=trailing-trivia`.  You're going to simply make that into `leading-trivia!` `=trailing-trivia` as the two tokens.